include/deal.II/fe/fe.h

```diff
@@ -1524,15 +1524,17 @@ class FiniteElement : public Subscriptor, public FiniteElementData<dim>
       ReferenceCell::default_combined_face_orientation()) const;
 
   /**
-   * For lines with non-standard line_orientation in 3d, the dofs on lines
-   * have to be permuted in order to be combined with the correct shape
-   * functions. Given a local dof @p index on a line, return the local index,
-   * if the line has non-standard line_orientation. In 2d and 1d there is no
-   * need for permutation, so the given index is simply returned.
+   * Given a local dof @p index on a line and the orientation @p
+   * combined_orientation of that line, return the local dof which accounts for
+   * @p combined_orientation.
+   *
+   * @note In both 1d and all-quadrilateral meshes in 2d all lines have the
+   * standard orientation.
```

Comment:
Is the latter actually true? It is true if you call the grid reorientation functions, but are they always called?

Reply:
We explicitly assume that this is true in some places (e.g., we don't compute some things in 2d) but this assumption doesn't hold for triangles and therefore shouldn't hold for mixed 2d meshes. We also have to look up values on flipped faces for computing periodicity constraints anyway so we shouldn't make this assumption.

Honestly, that particular assumption in the library is a bit of a mess and I want to remove it as part of these patches (it definitely won't hold any more when we have mixed meshes in 2d).

Reply:
That sounds good, thank you!